### PR TITLE
API returns 500 in case network is not found instead of 404

### PR DESCRIPTION
**Pull request by zhangguanzhang**
Signed-off-by: zhangguanzhang <zhangguanzhang@qq.com>
Fixes: https://github.com/containers/podman/issues/7184

### CODE REVIEW

File: pkg/api/handlers/compat/networks.go [hunk 1]
```diff
@@ -2,6 +2,7 @@ package compat
 
 import (
 	"encoding/json"
+	"github.com/containers/podman/v2/libpod/define"
```
Comment (orenbm) [inline pkg/api/handlers/compat/networks.go:5]: looks like this should go in the batch that starts on line 13

File: pkg/api/handlers/libpod/networks.go [hunk 2]
```diff
@@ -2,13 +2,13 @@ package libpod
 
 import (
 	"encoding/json"
+	"github.com/containers/podman/v2/libpod/define"
```
Comment (orenbm) [inline pkg/api/handlers/libpod/networks.go:5]: looks like this should go in the batch that starts on line 8

File: pkg/network/files.go [hunk 3]
```diff
@@ -3,6 +3,7 @@ package network
 import (
 	"encoding/json"
 	"fmt"
+	"github.com/containers/podman/v2/libpod/define"
```
Comment (orenbm) [inline pkg/network/files.go:6]: looks like this should go in the batch that starts on line 13

File: pkg/network/network.go [hunk 4]
```diff
@@ -2,6 +2,7 @@ package network
 
 import (
 	"encoding/json"
+	"github.com/containers/podman/v2/libpod/define"
```
Comment (orenbm) [inline pkg/network/network.go:5]: looks like this should go in the batch that starts on line 11

File: test/apiv2/35-networks.at [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# -*- sh -*-
+#
+# network-related tests
+#
+
+t GET /networks/non-existing-network 404
+
+# vim: filetype=sh
```
Comment (orenbm) [inline test/apiv2/35-networks.at:8]: missing newline at end of file

**Comment by zhangguanzhang** [comment]
@orenbm thanks

**Comment by zhangguanzhang** [comment]
@vrothberg PTAL

File: pkg/network/config.go [hunk 6]
```diff
@@ -20,10 +19,6 @@ const (
 	DefaultPodmanDomainName = "dns.podman"
 )
 
-var (
-	ErrNetworkNotFound = errors.New("network not found")
```
Comment (rhatdan) [inline pkg/network/config.go:24]: This should have been left and just a redirect to define.ErrNetowrkNotFound, otherwise you are changing the API.

**Pull request merged**